app/src/main/java/com/alexchan/wallpaper/ui/wallpaper/dashboard/dashboard/DashboardFragment.kt

```diff
@@ -0,0 +1,261 @@
+import androidx.recyclerview.widget.GridLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import androidx.recyclerview.widget.StaggeredGridLayoutManager
+import com.alexchan.wallpaper.R
+import com.alexchan.wallpaper.adapter.photoBinding.PhotoGridAdapter
+import com.alexchan.wallpaper.databinding.FragmentDashboardBinding
+import com.alexchan.wallpaper.ui.mainActivity.MainActivity
+import com.alexchan.wallpaper.ui.search.searchActivity.SearchActivity
+import com.alexchan.wallpaper.ui.wallpaper.wallpaper.WallpaperFragmentDirections
+import kotlinx.android.synthetic.main.activity_main.*
+import kotlinx.android.synthetic.main.fragment_dashboard.*
+
+
+class DashboardFragment : Fragment() {
+
+    private val dashboardViewModel: DashboardViewModel by lazy {
+        ViewModelProvider(this).get(DashboardViewModel::class.java)
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+        val binding = FragmentDashboardBinding.inflate(inflater)
+
+        // Allows Data Binding to Observe LiveData with the lifecycle of this Fragment
+        binding.lifecycleOwner = this
+
+        // Giving the binding access to the DashboardViewModel
+        binding.dashboardViewModel = dashboardViewModel
+
+        binding.photosGrid.adapter = PhotoGridAdapter(PhotoGridAdapter.OnClickListener {
+            dashboardViewModel.displayUserCollection(it)
+        })
+        dashboardViewModel.navigateToUserCollection.observe(viewLifecycleOwner, Observer {
+            if (null != it) {
+                this.findNavController().navigate(
+                    WallpaperFragmentDirections.actionShowDetail(it))
+                dashboardViewModel.displayUserCollectionComplete()
+            }
+        })
+
+        // Show Top Toolbar
+        requireActivity().topToolbar.visibility = View.VISIBLE
+
+        // Set TopToolbar OnMenuItemClickListener
+        requireActivity().topToolbar.setOnMenuItemClickListener{item: MenuItem? -> onMenuItemClick(item)}
+
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        // Reset Search Query
+        MainActivity.searchQuery = ""
+        MainActivity.searchStatus = false
+
+        // Reset Pagination
+        MainActivity.paginationStatus = true
+
+        val binding = DataBindingUtil.bind<FragmentDashboardBinding>(view)
+
+        // Use Shared Preference to get user selected display view type
+        val sharedPreferences: SharedPreferences = requireActivity().getSharedPreferences(
+            DISPLAY_VIEW_TYPE, Context.MODE_PRIVATE)
+
+        // Set Layout Manager Programmatically
+        when (sharedPreferences.getInt(DISPLAY_VIEW_TYPE_KEY, 2)) {
+            0 -> binding?.photosGrid?.layoutManager = StaggeredGridLayoutManager(1, StaggeredGridLayoutManager.VERTICAL)
```

Comment:
Can store this with share preferences, so when users come back they still see the same layout

Reply:
The above shared preferences works as follow, if this is the first time the user download and load the application, the default layout manager will be in Staggered Grid, after selecting any of the other display view type, it will stored in shared preferences, and the user will see the same layout as well as in the search results activity until the user decided to change to other display view type.

Reply:
This is the function that get executed when the user click on the change to list view display view type:
`getUserSelectedDisplayViewType(0)`

```
private fun getUserSelectedDisplayViewType(displayViewType: Int) {
        val sharedPreferences: SharedPreferences = requireActivity().getSharedPreferences(
            DISPLAY_VIEW_TYPE, Context.MODE_PRIVATE)
        val editor = sharedPreferences.edit()
        editor.putInt(DISPLAY_VIEW_TYPE_KEY, displayViewType)
        editor.apply()
    }
```

Reply:
Should make another class for sharePreferences. So this class can be tidier, and sharePreferences is easier to manage.